libp2p/src/main/kotlin/io/libp2p/pubsub/gossip/GossipRouter.kt

```diff
@@ -572,9 +590,14 @@ open class GossipRouter(
         enqueueIwant(peer, messageIds)
     }
 
+    private fun iDontWant(peer: PeerHandler, messageIds: List<MessageId>) {
+        if (messageIds.isEmpty()) return
+        enqueueIdontwant(peer, messageIds)
```

Comment:
The thing of `IDONTWANT` is that it needs to be flushed immediately to have the effect